server/src/test/java/io/whitefox/core/FakeFileSignerFactory.java

```diff
@@ -0,0 +1,13 @@
+package io.whitefox.core;
+
+import io.whitefox.core.services.FileSigner;
+import io.whitefox.core.services.FileSignerFactory;
+import io.whitefox.core.services.NoOpSigner;
+
+public class FakeFileSignerFactory implements FileSignerFactory {
```

Comment:
Probably you can build a single factory that returns both noop and s3 file signer based on the storage object passed it.